### PR TITLE
[Django 4.2]: Django 4.2 Upgrade

### PR DESCRIPTION
**Pull request by irtazaakram**
Upgrading to django42 and dropped django32 support.

- [x] verified stage and prod are using mysql8. 
- [x] make migrations executed
- [x] Removed django32 from tox and gha.
- [ ] Owning team will test and merge
- [ ] https://github.com/openedx/credentials/pull/2208 ( merge this )

- Resolves https://github.com/openedx/credentials/issues/2096

### CODE REVIEW

File: credentials/settings/base.py [hunk 1]
```diff
@@ -226,6 +226,9 @@
 TIME_ZONE = "UTC"
 TIME_ZONE_CLASS = timezone.utc
 
+# https://docs.djangoproject.com/en/4.2/releases/4.0/#zoneinfo-default-timezone-implementation
+USE_DEPRECATED_PYTZ = True
```
Comment (awais786) [inline credentials/settings/base.py:230]: Django switched from the `pytz` third-party python package to the Python native package `zoneinfo` in Django 4.0. In release notes, they have mentioned that both are not equivalent so one might need some manual work for this migration. So Django provided this `USE_DEPRECATED_PYTZ` setting, so we can decouple this migration from the Django upgrade.

More info: https://docs.djangoproject.com/en/4.2/releases/4.0/#zoneinfo-default-timezone-implementation

File: requirements/all.txt [hunk 2]
```diff
@@ -199,8 +202,9 @@ django-rest-swagger==2.2.0
     #   -r requirements/production.txt
 django-ses==3.5.0
     # via -r requirements/production.txt
-django-simple-history==3.1.1
+django-simple-history==3.0.0
```
Comment (awais786) [inline requirements/all.txt:205]: why downgrades ?

File: Makefile [hunk 3]
```diff
@@ -167,7 +167,7 @@ detect_changed_source_translations: ## Check if translation files are up-to-date
 	cd credentials && i18n_tool changed
 
 validate_translations: ## Test translations files
-	cd credentials && i18n_tool validate -v --check-all
+	cd credentials && i18n_tool validate --verbose
```
Comment (awais786) [inline Makefile:170]: It's giving some error due to latest version. Requested owning team to fix it. But need to continue with this PR so matching it like [platform](https://github.com/openedx/edx-platform/blob/master/Makefile#L54).

**Comment by awais786** [comment]
https://github.com/openedx/credentials/pull/2221 We are blocked due to translations check but this Temp PR is green using the fix hash. Hopefully we will get new version so i will update that in this PR.

File: requirements/production.txt [hunk 4]
```diff
@@ -65,17 +69,15 @@ cryptography==41.0.4
...
-defusedxml==0.8.0rc1
+defusedxml==0.8.0rc2
     # via
     #   -r requirements/base.txt
     #   python3-openid
     #   social-auth-core
 didkit==0.3.2
     # via -r requirements/base.txt
-django==3.2.21
+django==4.2.6
```
Comment (awais786) [inline requirements/production.txt:79]: Upgrading to `4.2.6`

**Review by awais786 — APPROVED** [review]

**Review by jsnwesson — APPROVED** [review]

**Pull request merged**